### PR TITLE
De-static GPGNetServer

### PR DESCRIPTION
**Pull request by Brutus5000**

### CODE REVIEW

File: ice-adapter/src/main/java/com/faforever/iceadapter/IceAdapter.java [hunk 1]
```diff
@@ -159,7 +145,7 @@ public static void close(int status) {
         log.info("close() - stopping the adapter. Status: {}", status);
```
Comment (GodFuper) [inline ice-adapter/src/main/java/com/faforever/iceadapter/IceAdapter.java:145]: You need to remove `INSTANCE.close();` as this generates recursion.

File: ice-adapter/src/main/java/com/faforever/iceadapter/gpgnet/GPGNetServer.java [hunk 2]
```diff
@@ -220,7 +238,7 @@ private static void onGpgnetConnectionLost() {
     /**
      * Listens for incoming connections from a game instance
      */
-    private static void acceptThread() {
+    private void acceptThread() {
         while (!Thread.currentThread().isInterrupted()) {
             log.info("Listening for incoming connections from game");
             try (Socket socket = serverSocket.accept()) {
```
Comment (GodFuper) [inline ice-adapter/src/main/java/com/faforever/iceadapter/gpgnet/GPGNetServer.java:244]: There's a mistake here. We close the socket immediately after executing try.

File: ice-adapter/src/main/java/com/faforever/iceadapter/gpgnet/GPGNetServer.java [hunk 3]
```diff
@@ -220,10 +238,14 @@ private static void onGpgnetConnectionLost() {
...
      */
-    private static void acceptThread() {
+    private void acceptThread() {
         while (!Thread.currentThread().isInterrupted()) {
             log.info("Listening for incoming connections from game");
-            try (Socket socket = serverSocket.accept()) {
+            try {
+                // The socket declaration must not be moved into a try-with-resources block, as the socket must not be
+                // closed. It is passed into the GPGNetClient.
+                Socket socket = serverSocket.accept();
```
Comment (GodFuper) [inline ice-adapter/src/main/java/com/faforever/iceadapter/gpgnet/GPGNetServer.java:247]: Maybe can create a constructor in the `GPGNetClient(ServerSocket serverSocket)` and get a socket in the constructor code. Assuming that the client should close the socket, and not try close in try/catch.

Comment (Brutus5000) [reply ice-adapter/src/main/java/com/faforever/iceadapter/gpgnet/GPGNetServer.java:247]: I am not even sure why it is split up in 2 classes. 1 class is probably sufficient and then keeps the reference to the socket. But that is out of scope of this PR.

File: ice-adapter/src/main/java/com/faforever/iceadapter/gpgnet/GPGNetServer.java [hunk 4]
```diff
@@ -21,51 +21,69 @@
...
         } else {
-            LOBBY_PORT = lobbyPort;
-            log.info("Using LOBBY_PORT: {}", LOBBY_PORT);
+            this.lobbyPort = lobbyPort;
+            log.info("Using LOBBY_PORT: {}", this.lobbyPort);
         }
 
-        try {
-            serverSocket = new ServerSocket(GPGNET_PORT);
+        try (ServerSocket Socket = new ServerSocket(gpgnetPort)) {
```
Comment (GodFuper) [inline ice-adapter/src/main/java/com/faforever/iceadapter/gpgnet/GPGNetServer.java:71]: closed too

**Comment by GodFuper** [comment]
Now you can merge

**Pull request merged**